stl/inc/ranges

```diff
@@ -3767,10 +3772,16 @@ namespace ranges {
             }
 
             _NODISCARD constexpr decltype(auto) operator*() const
-                noexcept(noexcept(_STD get<_Index>(*_Current))) /* strengthened */ {
+                noexcept(noexcept(_STD get<_Index>(*_Current))) /* strengthened */
+                requires is_reference_v<range_reference_t<_Base>> {
                 return _STD get<_Index>(*_Current);
             }
 
+            _NODISCARD constexpr decltype(auto) operator*() const noexcept(is_nothrow_move_constructible_v<_ElemTy> //
+                    && noexcept(_STD get<_Index>(*_Current))) /* strengthened */ {
+                return static_cast<remove_cv_t<_ElemTy>>(_STD get<_Index>(*_Current));

```

Comment:
`remove_cv_t<_ElemTy>` is `remove_cv_t<tuple_element_t<_Index, range_value_t<_Base>>>`. However, LWG-3502 depicts `remove_cv_t<tuple_element_t<N, range_reference_t<Base>>>`. As this is the `!is_reference_v<range_reference_t<_Base>>` case, I believe they're identical, but I wanted to double-check.

Reply:
I think they're the same as well, but (1) I see no reason to use `range_value_t` in some places and `range_reference_t` in others when we could be consistent, and (2) I'd hate to make every reader of this code work out their own proof of the equivalence.

I'll fix this issue since I introduced it.

Reply:
It's certainly possible and legitimate for a range to have a value type of `tuple<int>` and a reference type of `tuple<int&>`.